### PR TITLE
test(launchpad): add e2e tests for Project Setup - Round 2

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Apart of https://github.com/cypress-io/cypress/issues/19051

**Test Cases:**
- Project Setup (never setup e2e or CT before)
    - The compare testing types modal displays header, list, and code examples of testing types
    - Clicking a link for 'Need help' within the 'learn the difference' modal will open the docs link within the user's default browser
    - Testing type cards contain an icon that has a hover animation.
    - When no initial setup, the testing type cards read "Not Configured"
    - Clicking x on the 'learn the difference' modal will close the modal
    - Hitting enter key when focused on x on the 'learn the difference' modal will close the modal
    - Hitting escape on the 'learn the difference' modal will close the modal
    - Clicking off of the 'learn the difference' modal will close the modal
    
 - E2E Project Setup (Existing Project, Migrated, w/o E2E Configured)
   - Running `cypress open --e2e` with no initial setup detected will launch into the project in e2e showing the first step in configuration
   - Clicking on the e2e testing card of the project setup page will route the user to the Configuration Files page

- e2e Configured Files (never having setup E2E before)
  - Clicking 'Continue' takes you to the 'Choose a Browser' page

- component Project Setup (never having setup CT before)
    - Running `cypress open --component` with no initial setup detected will launch into the project in component showing 'choose framework' page
    - Clicking on the component area of project setup will route the user to the page showing 'choose framework' page

### User facing changelog
n/a

### PR Tasks
[n/a ] Have tests been added/updated?
[n/a] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
[n/a] Has a PR for user-facing changes been opened in cypress-documentation?
[n/a] Have API changes been updated in the type definitions?
[n/a] Have new configuration options been added to the cypress.schema.json?

### CODE REVIEW

File: packages/data-context/src/data/ProjectLifecycleManager.ts [hunk 1]
```diff
@@ -320,27 +321,33 @@ export class ProjectLifecycleManager {
       return
     }
 
-    if (!this.metaState.hasValidConfigFile) {
-      if (testingType === 'e2e' && !this.ctx.isRunMode) {
-        this.ctx.actions.wizard.scaffoldTestingType().catch(this.onLoadError)
```
Comment (emilyrohrbough) [inline packages/data-context/src/data/ProjectLifecycleManager.ts:325]: I assume that we need to add in a graceful error message for users that hit this when in run mode.

File: packages/data-context/src/actions/ProjectActions.ts [hunk 2]
```diff
@@ -85,7 +85,6 @@ export class ProjectActions {
 
   async setCurrentProject (projectRoot: string) {
     await this.clearCurrentProject()
```
Comment (emilyrohrbough) [inline packages/data-context/src/actions/ProjectActions.ts:87]: `this.clearCurrentProject()` already calls `this.ctx.lifecycleManager.clearCurrentProject()`: https://github.com/cypress-io/cypress/pull/19580/files#diff-d36f6626bf805d1bbbfba4786ca3a518f0083613a9814fd037c5592808949606L56

File: packages/launchpad/cypress/e2e/project-setup.cy.ts [hunk 3]
```diff
@@ -1,26 +1,271 @@
...
+    beforeEach(() => {
+      cy.openProject('pristine')
+      cy.visitLaunchpad()
+      verifyWelcomePage({ e2eIsConfigured: false, ctIsConfigured: false })
+    })
+
+    it('welcome page has "learn about testing types" link which opens modal', () => {
+      cy.contains('Review the differences').click()
+      cy.get('#app').should('have.attr', 'aria-hidden', 'true')
+      cy.contains('Key Differences').should('be.visible')
```
Comment (tbiethman) [inline packages/launchpad/cypress/e2e/project-setup.cy.ts:31]: You can get a tighter assertion by doing something like:
```js
cy.findByRole('dialog', { name: 'Key Differences' }.within(() => {
  // assert scoped dialog content here
})
```

It'd be nice to dry up things like the app being hidden when a dialog is open and other common dialog behaviors (like the ever-present "Need help?" link). Custom command maybe? Not necessary to change here but worth some thinking.

```js
cy.getDialog({ name: '...', rootSelector: '#app' }) // asserts aria state common dialog features -> returns dialog element 
.within()
```

Comment (emilyrohrbough) [reply packages/launchpad/cypress/e2e/project-setup.cy.ts:31]: Good point. I was finding cy.contains would match a model on the dom even if it wasn't open. I'll make the updates.

File: packages/launchpad/cypress/e2e/support/e2eSupport.ts [hunk 4]
```diff
@@ -2,3 +2,4 @@
 require('../../../../frontend-shared/cypress/e2e/support/e2eSupport')
 
 require('./dropFileWithPath')
+require('cypress-plugin-tab')
```
Comment (tbiethman) [inline packages/launchpad/cypress/e2e/support/e2eSupport.ts:5]: :eyes: did some reading about this, interesting. We've also [recommended](https://github.com/cypress-io/cypress/issues/311#issuecomment-787508664) https://github.com/dmtrKovalenko/cypress-real-events previously, but there's no firefox support there

Comment (emilyrohrbough) [reply packages/launchpad/cypress/e2e/support/e2eSupport.ts:5]: Yeah...We have is issue logged https://github.com/cypress-io/cypress/issues/299 in the backlog for tab support so hopefully we can implement & provide first-class support soon.

**Review by tbiethman — APPROVED** [review]

**Review by ZachJW34 — APPROVED** [review]

**Pull request merged**